docs/how-to-guides/develop-a-tool/use-cascading-inputs-in-tool.md

```diff
@@ -0,0 +1,136 @@
+# Use Cascading Inputs in Tool
+
+Cascading settings between inputs are frequently used in situations where a selection in one input field determines what subsequent inputs should be shown. This approach helps create a more efficient, user-friendly, and error-free input process. This article will guide you through the process of implementing cascading settings for tool inputs.
+
+## Prerequisites
+Please ensure that your [Prompt flow for VS Code](https://marketplace.visualstudio.com/items?itemName=prompt-flow.prompt-flow) is updated to version 1.2.0 or later.
+
+
+## Create tool with cascading inputs
+We will provide a hands-on tool example to demonstrate the implementation of cascading settings between inputs within a tool. 
+In this tool, the availability of 'student_id' and 'teacher_id' inputs is determined by the value of the 'user_type' input. 
+Below shows how to support this cascading setting in both tool code and tool yaml.
+
+1. Develop your tool within the def function, referring to [tool_with_cascading_inputs.py](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_cascading_inputs.py) as an example. You need to pay attention to some key points:
+    * Use the `@tool` decorator to identify the function as a tool.
+    * When an input should only take on a certain set of fixed values, an Enum class such as `UserType` as shown in the following example can be created.
+    * Within the tool's logic, various inputs may be used depending on the value of the `user_type` input.
+
+```python
+from enum import Enum
+
+from promptflow import tool
+
+
+class UserType(str, Enum):
+    STUDENT = "student"
+    TEACHER = "teacher"
+
+
+@tool
+def my_tool(user_type: Enum, student_id: str = "", teacher_id: str = "") -> str:
+    """This is a dummy function to support cascading inputs.
+
+    :param user_type: user type, student or teacher.
+    :param student_id: student id.
+    :param teacher_id: teacher id.
+    :return: id of the user.
+    If user_type is student, return student_id.
+    If user_type is teacher, return teacher_id.
+    """
+    if user_type == UserType.STUDENT:
+        return student_id
+    elif user_type == UserType.TEACHER:
+        return teacher_id
+    else:
+        raise Exception("Invalid user.")
+```
+
+2. Follow the guide [Create and Use Tool Package](create-and-use-tool-package.md) to generate a tool yaml for your tool. Afterwards, manually update this tool yaml to transition from common inputs to cascading inputs.
```

Comment:
Generate a starting YAML for your tool per the [tool package guide](https://claude.ai/chat/create-and-use-tool-package.md), then update it to enable cascading:

Add "enabled_by" and "enabled_by_value" to control visibility of dependent inputs.
See the [example YAML](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/yamls/tool_with_cascading_inputs.yaml) for reference.
Note: "enabled_by_value" takes a list, allowing multiple values to enable an input.